brokers/theia/sidecar.go

```diff
@@ -53,7 +54,11 @@ func AddExtension(toolingConf *model.ToolingConf, pj model.PackageJSON) {
 	sidecarEndpoint := toolingConf.Endpoints[0]
 	prettyID := re.ReplaceAllString(pj.Publisher+"_"+pj.Name, `_`)
 	sidecarTheiaEnvVarName := "THEIA_PLUGIN_REMOTE_ENDPOINT_" + prettyID
-	sidecarTheiaEnvVarValue := "ws://" + sidecarEndpoint.Name + ":" + strconv.Itoa(sidecarEndpoint.TargetPort)
+	sidecarHostname := sidecarEndpoint.Name
+	if cfg.UseLocalhostInPluginUrls {
```

Comment:
When I was designing this utility it was supposed to not interact with config package. In fact, interacting with the config package is like using global variables.

Reply:
That's what it is: a global variable that is taken into account every time we build an endpoint URL. BTW this global setting is only changed when the brokers are used as a library, so no change for your use-case.

Reply:
Here I'm more concerned about maintainability of the codebase rather than any particular case

Reply:
Maybe this concern could be tackled in the context of one of the future PRs that are still expected in the upcoming refactoring work on the `che-plugin-broker` ?

I tried to reduce the changes in this PR to the minimum, and in a way that introduce only local non-invasive changes.